chainstate/test-suite/src/tests/nft_transfer.rs

```diff
@@ -0,0 +1,222 @@
+use chainstate::{BlockError, ChainstateError, TokensError};
+use chainstate::{CheckBlockError, CheckBlockTransactionsError, ConnectTransactionError};
+use chainstate_test_framework::{TestBlockInfo, TestFramework, TransactionBuilder};
+use common::chain::{
+    signature::inputsig::InputWitness,
+    tokens::{
+        token_id, Metadata, NftIssuanceV1, OutputValue, TokenCreator, TokenData, TokenId,
+        TokenTransferV1,
+    },
+    Destination, OutputPurpose, TxInput, TxOutput,
+};
+use common::primitives::Amount;
+use crypto::key::{KeyKind, PrivateKey};
+use crypto::random::distributions::uniform::SampleRange;
+use crypto::random::Rng;
+use rstest::rstest;
+use test_utils::random::{make_seedable_rng, Seed};
+
+// FIXME(nft_issuance): This is the copy of function from check block. Remove copy and use this func from more appropriate place.
+fn random_creator() -> TokenCreator {
+    let (_, public_key) = PrivateKey::new(KeyKind::RistrettoSchnorr);
+    TokenCreator::from(public_key)
+}
+
+//FIXME(nft_issuance): Move it in super mod and use for all tokens tests
+fn random_string<R: SampleRange<usize>>(rng: &mut impl Rng, range_len: R) -> String {
+    use crypto::random::distributions::{Alphanumeric, DistString};
+    if range_len.is_empty() {
+        return String::new();
+    }
+    let len = rng.gen_range(range_len);
+    Alphanumeric.sample_string(rng, len)
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn nft_transfer_test(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut tf = TestFramework::default();
+        let mut rng = make_seedable_rng(seed);
+        // To have possibility to send exceed tokens amount than we have, let's limit the max issuance tokens amount
```

Comment:
What does this mean?

Reply:
It's still developing and some code just a copy of the fungible tokens. The PR is still in draft.